### PR TITLE
Switched to using staticcheck linter.

### PR DESCRIPTION
**Pull request by mwhittaker**
Previously, we used [golangci-lint](https://github.com/golangci/golangci-lint), but the Go team recommended we use staticcheck instead.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -41,12 +41,12 @@ jobs:
       - name: Cache linter
         uses: actions/cache@v3
         with:
-          path: ~/go/bin/golangci-lint
-          key: golangci-lint-v1.51.2
+          path: ~/go/bin/staticcheck
+          key: staticcheck-v0.4.3
```
Comment (ghemawat) [inline .github/workflows/go.yml:45]: Can you extend the commit message to say why we are doing this and what the pros and cons are for staticcheck vs golangci-lint.

File: internal/benchmarks/benchmarks_test.go [hunk 2]
```diff
@@ -362,10 +361,6 @@ func BenchmarkPing(b *testing.B) {
 	}
 }
 
-func init() {
-	rand.Seed(time.Now().UnixNano())
-}
-
```
Comment (ghemawat) [inline internal/benchmarks/benchmarks_test.go:368]: Why are we removing this seed initialization?

File: staticcheck.conf [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# See https://staticcheck.dev/docs/configuration/options/#checks.
+checks = ["all", "-ST1000", "-ST1003", "-ST1005", "-ST1012", "-ST1016", "-ST1020", "-ST1021", "-ST1022"]
```
Comment (ghemawat) [inline staticcheck.conf:2]: Can we add comments saying what the dropped checks are and why we are dropping them?  I think some of them are dropped to match the default value, but some are new.

**Review by ghemawat — COMMENTED** [review]
Thanks!

**Review by mwhittaker — COMMENTED** [review]
Thanks Sanjay!

File: staticcheck.conf [hunk 4]
```diff
@@ -1,2 +1,17 @@
-# See https://staticcheck.dev/docs/configuration/options/#checks.
+# In addition to the default checks [1], we also disable the following style
```
Comment (ghemawat) [inline staticcheck.conf:1]: How about:
  In addition to the checks disabled by default [1], we also ...

**Review by ghemawat — APPROVED** [review]
Drop the "financially ..." part from the change description. Doesn't seem relevant to what checker we pick.

File: staticcheck.conf (hunk 4 above)
Comment (mwhittaker) [reply staticcheck.conf]: Much clearer, thanks!

**Review by mwhittaker — COMMENTED** [review]
> Drop the "financially ..." part from the change description. Doesn't seem relevant to what checker we pick.

Got it, will do.

**Pull request merged**